packages/debugger/src/model.ts

```diff
@@ -164,13 +214,38 @@ export class DebuggerModel implements IDebugger.Model.IService {
     this.title = '-';
   }
 
+  filterVariablesByViewOptions(
+    variables: IDebugger.IVariable[],
+    variablesFilterOptionsMap: Map<VariablesFilterOptionKey, boolean>
+  ): IDebugger.IVariable[] {
+    let filteredVariables = variables;
+    for (const [key, enabled] of variablesFilterOptionsMap) {
+      if (enabled) {
+        const viewFilter =
+          variablesFilterOptions[key as VariablesFilterOptionKey]?.filter;
+        if (viewFilter) {
+          filteredVariables = filteredVariables.filter(viewFilter);
+        }
+      }
+    }
+    return filteredVariables;
+  }
+
   private _disposed = new Signal<this, void>(this);
   private _isDisposed = false;
   private _hasRichVariableRendering = false;
   private _supportCopyToGlobals = false;
   private _stoppedThreads = new Set<number>();
   private _title = '-';
   private _titleChanged = new Signal<this, string>(this);
+  private _variablesFilterOptionsChanged = new Signal<
+    this,
+    Map<VariablesFilterOptionKey, boolean>
+  >(this);
+  private _variablesFilterOptions = new Map<
```

Comment:
Is there any advantage of using a `Map` instead of simple object ? Note that I don't have opinion on it, only wonder.